src/download.rs

```diff
@@ -242,13 +245,26 @@ pub fn download_file(
     match pb {
         Some(ref pb) => {
             copy_largebuf(&mut pb.wrap_read(response), &mut buffer)?;
-            pb.finish_with_message("Done");
+            pb.finish();
         }
         None => {
             copy_largebuf(&mut response, &mut buffer)?;
         }
     }
 
+    let downloaded_length: u64;
+    if let Some(total_length) = total_length {
+        downloaded_length = total_length - starting_length;
+    } else {
+        downloaded_length = fs::metadata(dest_name).unwrap().len();
+    }
```

Comment:
`fs::metadata()` won't work if we're printing to stdout, because then `dest_name` isn't a real file.

We shouldn't print anything if `quiet` is true (and then we also don't have a progress bar).

So maybe:
```suggestion
    let downloaded_length = pb.position() - starting_length;
```
and then move the whole thing into the `Some(ref pb)` block above?

[`.finish()` updates the position](https://docs.rs/indicatif/0.15.0/src/indicatif/progress.rs.html#631), so maybe it's best to compute `downloaded_length` before calling `.finish()`.